packages/cms-data-sync/src/calendars/sync-all.ts

```diff
@@ -0,0 +1,46 @@
+import { Logger } from '@asap-hub/server-common';
+import { CalendarDataProvider } from '@asap-hub/model';
+
+import { fetchContentfulEntries, getContentfulClient } from '../utils';
+import { rateLimiter } from '../utils/rate-limiter';
+
+export async function syncCalendars({
+  dataProvider,
+  logger,
+}: {
+  dataProvider: CalendarDataProvider;
+  logger: Logger;
+}) {
+  const { contentfulEnvironment } = await getContentfulClient();
+
+  const calendars = await fetchContentfulEntries(
+    contentfulEnvironment,
+    'calendars',
+  );
+
+  if (!calendars.length) {
+    logger.info('No calendars to sync');
+    return;
+  }
+
+  const promises = await Promise.allSettled(
+    calendars.map(async (calendar) => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      await dataProvider.update(calendar.sys.id, { googleApiMetadata: {} });
+      await rateLimiter.removeTokens(3);
+    }),
+  );
+
+  promises.forEach((promise, i) => {
+    if (promise.status === 'rejected') {
+      const calendar = calendars[i];
+      logger.error(
+        promise.reason,
+        `Failed to sync calendar: ${calendar.sys.id}`,
+      );
+    }
+  });
```

Comment:
Again we probably do want to throw errors here if syncing fails rather than swallowing them.

Reply:
Would it make sense to log all errors then if there has been an error throw an error. So we get all errors if there are multiple?